package/samplers/synetune_sampler/syne_tune_sampler.py

```diff
@@ -0,0 +1,142 @@
+                self._syne_tune_space, metric=self.metric, mode=self.mode, **searcher_kwargs
+            )
+        )
+
+    def _convert_optuna_to_syne_tune(
+        self, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, Any]:
+        syne_tune_space = {}
+        for name, distribution in search_space.items():
+            if isinstance(distribution, CategoricalDistribution):
+                syne_tune_space[name] = choice(distribution.choices)
+            elif distribution.step is not None:
+                if isinstance(distribution, IntDistribution):
+                    syne_tune_space[name] = finrange(
+                        lower=distribution.low,
+                        upper=distribution.high,
+                        size=distribution.step,
+                        cast_int=True,
+                    )
+                elif isinstance(distribution, FloatDistribution):
+                    syne_tune_space[name] = finrange(
+                        lower=distribution.low,
+                        upper=distribution.high,
+                        size=distribution.step,
+                        cast_int=False,
+                    )
+            elif distribution.step is None and not distribution.log:
+                if isinstance(distribution, IntDistribution):
+                    syne_tune_space[name] = randint(
+                        lower=distribution.low, upper=distribution.high
+                    )
+                elif isinstance(distribution, FloatDistribution):
+                    syne_tune_space[name] = uniform(
+                        lower=distribution.low, upper=distribution.high
+                    )
+            elif distribution.log:
+                if isinstance(distribution, FloatDistribution):
+                    syne_tune_space[name] = loguniform(distribution.low, distribution.high)
+                else:
+                    syne_tune_space[name] = lograndint(distribution.low, distribution.high)
+            else:
+                raise NotImplementedError(f"Unknown Hyperparameter Type: {type(distribution)}")
+        return syne_tune_space
+
+    def sample_relative(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        search_space: dict[str, BaseDistribution],
+    ) -> dict[str, Any]:
+        trial_suggestion: Trial = self.scheduler.ask()
+        self.trial_mapping[trial_suggestion.trial_id] = trial_suggestion
+        return trial_suggestion.config
+
+    def after_trial(
+        self, study: Study, trial: FrozenTrial, state: TrialState, values: Sequence[float]
+    ) -> None:
+        params = trial.params
+
+        trial = Trial(
+            trial_id=trial.number,
+            config=params,
+            creation_time=self.trial_mapping[trial.number].creation_time,
+        )
+        experiment_results = {self.metric: values[0]}
+
+        self.scheduler.tell(trial=trial, experiment_result=experiment_results)
```

Comment:
Are there any reasons why the trial stored in `trial_mapping` was not used in the original implementation?

```suggestion
        self.scheduler.tell(trial=self.trial_mapping[trial.number], experiment_result={self.metric: values[0]})
```